File: LaboratoryWork/Semester_2/Laboratory_7/Laboratory_2/Java/Task3.java
import java.util.Scanner;

public class Task3 {
    public static boolean is_number(int a) {
        return (a / 10 > 0);
    }


    public static int cnt_odd_digits(int a) {
        int res = 0;
        a = Math.abs(a);

        while (a > 0) {
            res += (a % 2);
            a /= 10;
        }
        return res;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Введите количество чисел: ");
        int n = scanner.nextInt();


        for (int i = 0; i < n; i++) {
            System.out.print("Введите " + (i + 1) + " число: ");
            int num = scanner.nextInt();

            int cnt_odd = 0;
            if (is_number(num)) {
                cnt_odd = cnt_odd_digits(num);
            } else {
                System.out.println(num + " - это цифра!");
            }

            System.out.println("Количество нечетных цифр: " + cnt_odd);
        }
        scanner.close();    
    }
}

File: LaboratoryWork/Semester_2/Laboratory_2/Go/task3.go
package main

import (
	"fmt"
	"math"
)


func is_number(a int) bool {
	/* Проверяет, что это число, а не цифра. */
	return (a / 10 != 0)
}


func cnt_odd_digits(a int) int {
	/* Считает количество нечетных цифр в числе. */
	res := 0
	a = int(math.Abs(float64(a)))

	for a > 0 {
		res += a % 2
		a /= 10
	}
	return res
}


func task3() {
	var n int
	fmt.Print("Введите количество чисел: ")
	fmt.Scan(&n)

	for i := 0; i < n; i++ {
		var num int
		fmt.Print("Введите ", i + 1, " число: ")
		fmt.Scan(&num)

		cnt_odd := 0
		if is_number(num) {
			cnt_odd = cnt_odd_digits(num)
		} else {
			fmt.Println(num, " - это не цифра!")
		}

		fmt.Println("Количество нечетных цифр: ", cnt_odd)
	}
}